### ISSUE TITLE
Children prop type missing in HeadProvider

### ISSUE THREAD

**Issue by itaditya** [tizmagik/react-head#140]
Getting this TS error

<img width="950" alt="image" src="https://user-images.githubusercontent.com/15871340/171702422-daab0220-dfc8-464b-98d6-e379cfc3226f.png">

> Type '{ children: Element; }' has no properties in common with type 'IntrinsicAttributes & { headTags?: ReactElement<unknown, string | JSXElementConstructor<any>>[] | undefined; }'.

Fix for this-

since HeadProvider uses `children` prop, it's typescript definition should include `children: ReactNode`

**Comment by tizmagik** [comment]
Can you create a reproduction on Codesandbox or something? I'm not able to repro... https://codesandbox.io/s/modest-matsumoto-try7wg?file=/src/App.tsx

**Comment by itaditya** [comment]
here's a simple Vite based app. Let me know if this will do otherwise I'll make one on Stackblitz. 

https://github.com/itaditya/friendly-ui/blob/main/packages/react/src/App.tsx

**Comment by itaditya** [comment]
I noticed the types in your CSB are of React 17 whereas mine are of React 18


  "dependencies": {
    "react": "18.0.0",
    "react-dom": "18.0.0",
    "react-head": "3.4.0",
    "react-scripts": "4.0.3"
  },

**Comment by tizmagik** [comment]
Ah, yea I haven't tested react-head against React 18. It's possible it needs some updates. Feel free to open a PR to address any type issues with React 18.

**Comment by abhishiv** [comment]
Can confirm I face the same issue with react18. 

```
    const H = HeadProvider as any;
    const element = (
      <H headTags={headTags}>
      </H>
    );
```

**Comment by marhaupe** [comment]
This MR should fix this issue: https://github.com/tizmagik/react-head/pull/146

**Comment by tizmagik** [comment]
Fixed in [v3.4.2](https://github.com/tizmagik/react-head/releases/tag/v3.4.2)
